pkg/upgrade/upgrade.go

```diff
@@ -323,9 +324,10 @@ func UpdateFromLegacyVersion(cli client.Client, platform deploy.Platform, appNS
 			if err := deleteResource(cli, montNamespace, "statefulset"); err != nil {
 				return err
 			}
-			if err = CreateDefaultDSC(cli, platform); err != nil {
```

Comment:
I am not sure about this change. For non-upgrade cases, i.e., when no kfdefs are found, the creation of the DSC is handled by the code outside of this upgrade function. Can you please review this?

Reply:
can you point me to the code?

i was confused thought it should be there to created the DSC CR but i can only see this function call and is wrapped by the kfdef 
when i was testing with the current code base, after installed operator, only DSCI was created but not DSC.
then i tested the PR, it created the DSC automatically.

Reply:
I do not believe we should create the DSC automatically for new installs. Only for upgrades from previous versions. Was there a change in requirements?

Reply:
i've reverted this change